### PR TITLE
refactor src/publish/index to use async.waterfall

### PR DESCRIPTION
**Pull request by EndangeredMassa**
I can't test this on airplane wifi. This is a start to the async.waterfall refactor.

### CODE REVIEW

File: src/publish/index.coffee [hunk 1]
```diff
@@ -13,85 +14,79 @@ test = require './test'
...
+      tempChangelogPath = "/tmp/npub/changelog-#{sha}.md"
+      changelog.write(tempChangelog, tempChangelogPath)
+      done(null, tempChangeLog)
 
-                changelog.update(tempChangelogPath)
+    (tempChangelogPath, done) ->
+      openEditor tempChangelogPath, (error) ->
+        if error?
+          fs.unlinkSync tempChangelogPath
+        changelog.update(tempChangelogPath)
```
Comment (jkrems) [inline src/publish/index.coffee]: Is there a reason to call `changelog.update` when the editor failed? Same for `updateVersion`?

Comment (EndangeredMassa) [inline src/publish/index.coffee]: updated

**Comment by jkrems** [comment]
LGTM

**Comment by jkrems** [comment]
Wow, typing `:happy` auto-completes to `:smile:`. Github really has this emoji thing figured out.

**Comment by EndangeredMassa** [comment]
Actually tested this, fixed a couple issues, and now it works!

Brought back the git branch checking and pushing. I'm not sure why I thought that wasn't necessary before. We need to update the current branch remotely.

**Comment by jkrems** [comment]
:+1:

**Comment by EndangeredMassa** [comment]
refactor rebased; ready for review

**Comment by jkrems** [comment]
:+1:

**Pull request merged**